tests/writing.rst

```diff
@@ -1,6 +1,28 @@
 Writing library tests
 =====================
 
+.. note:: Generally we don't "unit test" the lower levels of Xapian
+
+   A common convention in a lot of software is to write `unit tests`_,
+   testing individual units of source code. In Xapian we generally
+   do not test the lower levels separately. Instead, we write tests
+   that use Xapian's public API.
+
+   So for instance, we don't write unit tests for our ``::Internal``
+   classes. So if you make a change to :class:`Xapian::Weight::Internal`
+   you wouldn't write a test for that change directly. However, your
+   changes are motivated by a desire to test behaviour in the library,
```

Comment:
I think `test` here isn't the word you meant, but not exactly sure what it should be - something like "to implement behaviour"?

Reply:
Umm, good point that makes no sense. How about now?

Reply:
Looks good.